test/component/agent_api_test.go

```diff
@@ -135,7 +136,7 @@ func TestMetrics(t *testing.T) {
 
 	agentAPI := plugins.NewAgentAPI(conf, mockEnvironment, mockNginxBinary)
 	agentAPI.Init(core.NewMockMessagePipe(context.TODO()))
-	agentAPI.Process(core.NewMessage(core.MetricReport, &proto.MetricsReport{
+	agentAPI.Process(core.NewMessage(core.MetricReport, &metrics.MetricsReportBundle{Data: []*proto.MetricsReport{{
```

Comment:
Can you update the component test to have more than one report in the MetricsReportBundle?

So like move the `system.cpu.idle` metric to a SYSTEM report and the `nginx.workers.count` to an INSTANCE report

Reply:
wild